kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/storageApi/StorageWriteApiWriter.java

```diff
@@ -121,5 +123,26 @@ public Runnable build() {
             }
             return new StorageWriteApiWriter(tableName, streamWriter, records, streamName);
         }
+
+        private JSONObject getJsonFromMap(Map<String, Object> map) {
+            JSONObject jsonObject = new JSONObject();
+            map.forEach((key, value) -> {
+                if (value instanceof Map<?, ?>) {
+                    value = getJsonFromMap((Map<String, Object>) value);
```

Comment:
@b-goyal Is it possible that the **value** here is an instance of anything but **Map<String, Object>**. In that case, I think the typecast won't work.

Reply:
No @sp-gupta , that should not be possible. You may take a look at the checks and map conversion logic in BigQueryRecordConverter. First that class performs conversion and then we receive the map object here.